intranet/apps/announcements/notifications.py

```diff
@@ -113,9 +114,15 @@ def announcement_posted_email(request, obj, send_all=False):
     if settings.EMAIL_ANNOUNCEMENTS:
         subject = "Announcement: {}".format(obj.title)
         if send_all:
-            users = get_user_model().objects.all()
+            users = get_user_model().objects.filter(
+                user_type__in=["student", "teacher", "counselor"], graduation_year__lte=get_senior_graduation_year()
```

Comment:
Wait... sorry (might have been my fault) but wouldn't this exclude all juniors, sophomores, and freshmen?

i.e. if `get_senior_graduation_year()` is 2021, then 2022 is not less than nor equal to 2021, so juniors would be excluded?

```suggestion
                user_type__in=["student", "teacher", "counselor"], graduation_year__gte=get_senior_graduation_year()
```

This would, though, exclude `teacher`s and `counselor`s, because they do not have `graduation_year`s, so maybe we would have to do a union?

Reply:
So maybe something like:
```
users = get_user_model().objects.filter(user_type="student", graduation_year__gte=get_senior_graduation_year()).union(get_user_model().objects.filter(user_type__in=["teacher", "counselor"]))
```
and then add the extra filter on there for the `else` statement.

Reply:
Yeah, I think that will work.